src/runtime_src/driver/xclng/tools/xbutil/xbutil.cpp

```diff
@@ -1479,3 +1493,174 @@ int xcldev::xclP2p(int argc, char *argv[])
+        return -ENOMEM;
+    }
+    boptr = (char*) xclMapBO(handle, boh, true);
+    if (boptr == nullptr) {
+        std::cout << "Error mapping BO" << std::endl;
+        m2m_free_unmap_bo(handle, boh, boptr, boSize);
+        return -EINVAL;
+    }
+    memset(boptr, pattern, boSize);
+    return boh;
+}
+static int m2mtest_bank(xclDeviceHandle handle, uuid_t uuid, int bank_a, int bank_b)
+{
+    unsigned boSrc = NULLBO;
+    unsigned boTgt = NULLBO;
+    char *boSrcPtr = nullptr;
+    char *boTgtPtr = nullptr;
+
+    const size_t boSize = 256L * 1024 * 1024;
+    if (xclOpenContext(handle, uuid, -1, true)) {
+        std::cout << "ERROR: Unable to lockdown xclbin" << std::endl;
+        return -EINVAL;
+    }
+
+    //Allocate and init boSrc
+    boSrc = m2m_alloc_init_bo(handle, boSrcPtr, boSize, bank_a, 'A');
+    if(xclSyncBO(handle, boSrc, XCL_BO_SYNC_BO_TO_DEVICE, boSize, 0)) {
+        std::cout << "ERROR: Unable to sync source BO" << std::endl;
+        m2m_free_unmap_bo(handle, boSrc, boSrcPtr, boSize);
+        return -EINVAL;
+    }
+
+    //Allocate and init boTgt
+    boTgt = m2m_alloc_init_bo(handle, boTgtPtr, boSize, bank_b, 'B');
+    if(xclSyncBO(handle, boTgt, XCL_BO_SYNC_BO_TO_DEVICE, boSize, 0)) {
+        std::cout << "ERROR: Unable to sync target BO" << std::endl;
+        m2m_free_unmap_bo(handle, boSrc, boSrcPtr, boSize);
+        m2m_free_unmap_bo(handle, boTgt, boTgtPtr, boSize);
+        return -EINVAL;
+    }
+    //Allocate the exec_bo
+    unsigned execHandle = xclAllocBO(handle, sizeof (ert_start_copybo_cmd),
+        xclBOKind(0), (1<<31));
+    struct ert_start_copybo_cmd *execData =
+        reinterpret_cast<struct ert_start_copybo_cmd *>(
+        xclMapBO(handle, execHandle, true));
+    ert_fill_copybo_cmd(execData, boSrc, boTgt, 0, 0, boSize);
+
+    xcldev::Timer timer;
+    if(xclExecBuf(handle, execHandle)) {
+        std::cout << "ERROR: Unable to issue xclExecBuf" << std::endl;
+        return -EINVAL;
+    }
+
+    while (execData->state < ERT_CMD_STATE_COMPLETED){
+        while (xclExecWait(handle, 1000) == 0) {
+            std::cout << "reentering wait...\n";
+        };
+    }
+    double timer_stop = timer.stop();
+
+    if(xclSyncBO(handle, boTgt, XCL_BO_SYNC_BO_FROM_DEVICE, boSize, 0)) {
+        m2m_free_unmap_bo(handle, boSrc, boSrcPtr, boSize);
+        m2m_free_unmap_bo(handle, boTgt, boTgtPtr, boSize);
+        m2m_free_unmap_bo(handle, execHandle, execData, sizeof (ert_start_copybo_cmd));
+        std::cout << "ERROR: Unable to sync target BO" << std::endl;
+        return -EINVAL;
+    }
+    
+    bool match = (memcmp(boSrcPtr, boTgtPtr, boSize) == 0);
+
+    // Clean up
+    m2m_free_unmap_bo(handle, boSrc, boSrcPtr, boSize);
+    m2m_free_unmap_bo(handle, boTgt, boTgtPtr, boSize);
+    m2m_free_unmap_bo(handle, execHandle, execData, sizeof (ert_start_copybo_cmd));
+
+    xclCloseContext(handle, uuid, -1);
+
+    if (!match) {
+        return -EINVAL;
```

Comment:
need error msg.

Reply:
"ERROR: xbutil m2mtest failed." is printed if the test fails